coda/coda_mdstore/tests/test_views.py

```diff
@@ -0,0 +1,890 @@
+
+    def test_delete_request(self, rf):
+        node = NodeFactory.create()
+        request = rf.delete('/', HTTP_HOST='example.com')
+        response = views.app_node(request, node.node_name)
+        assert response.status_code == 200
+
+    def test_delete_request_returns_not_found(self, rf):
+        request = rf.delete('/', HTTP_HOST='example.com')
+        response = views.app_node(request, 'DNE-001')
+        assert response.status_code == 404
+
+
+class TestAppBag:
+    """
+    Tests for coda_mdstore.views.app_bag.
+    """
+
+    def test_get_request_returns_not_found(self, rf):
+        request = rf.get('/', HTTP_HOST='example.com')
+        response = views.app_bag(request, 'ark:/00002/id')
+        assert response.status_code == 404
+
+    def test_get_request(self, rf):
+        FullBagFactory.create_batch(10)
+
+        request = rf.get('/', HTTP_HOST='example.com')
+        response = views.app_bag(request)
+
+        assert response.status_code == 200
+        assert response['Content-Type'] == 'application/atom+xml'
+
+    def test_get_request_response_content(self, rf):
+        FullBagFactory.create_batch(10)
+
+        request = rf.get('/', HTTP_HOST='example.com')
+        response = views.app_bag(request)
+
+        tree = objectify.fromstring(response.content)
+        assert len(tree.entry) == 10
+
+    @pytest.mark.xfail(reason="Refactor required.")
+    def test_get_request_with_identifier(self):
+        assert 0
+
+    @mock.patch('coda_mdstore.views.createBag')
+    def test_post_request(self, mock_createBag, rf):
+        bag = FullBagFactory.create()
+        mock_createBag.return_value = bag, bag.bag_info_set
+
+        request = rf.post('/', HTTP_HOST='example.com')
+        response = views.app_bag(request)
+
+        assert response.status_code == 201
+        assert response['Content-Type'] == 'application/atom+xml'
+        assert response['Location'] == (
+            'http://example.com/APP/bag/{0}/'.format(bag.name)
+        )
+
+    def test_delete_request_is_successful(self, rf):
+        bag = FullBagFactory.create()
+        request = rf.delete('/', HTTP_HOST='example.com')
+        response = views.app_bag(request, bag.name)
+
+        assert response.status_code == 200
+        assert models.Bag.objects.exists() is False
+
+    def test_delete_request_returns_bad_request(self, rf):
+        request = rf.delete('/', HTTP_HOST='example.com')
+        response = views.app_bag(request)
+        assert response.status_code == 400
+
+    def test_delete_request_removes_member_ext_identifier_objects(self, rf):
+        bag = FullBagFactory.create()
+        request = rf.delete('/', HTTP_HOST='example.com')
+
+        assert models.External_Identifier.objects.exists() is True
+
+        response = views.app_bag(request, bag.name)
+        assert response.status_code == 200
+        assert models.External_Identifier.objects.exists() is False
```

Comment:
I turned `test_delete_request_returns_bad_request` into a general test that will check that a HttpResponseBadRequest is returned.